internal/controller/operator/factory/vmauth/vmauth.go

```
@@ -119,6 +125,34 @@ func CreateOrUpdate(ctx context.Context, cr *vmv1beta1.VMAuth, rclient client.Cl
 	return nil
 }
 
+func createOrUpdateHTTPRoute(ctx context.Context, rclient client.Client, cr, prevCr *vmv1beta1.VMAuth) error {
+	if cr.Spec.HTTPRoute == nil {
+		return nil
+	}
+
+	if err := k8stools.GetCRD(ctx, rclient, "httproutes.gateway.networking.k8s.io"); err != nil {
+		if k8serrors.IsNotFound(err) {
```

> if CreateOrUpdate returns error it will automatically set status to Failed. Just need to write proper log message and text in error message

@AndrewChubatiuk So if vmauth.spec.HTTPRoute is not nil but the HTTPRoute CRD is missing, we should return an error in this case, correct?